### ISSUE TITLE
TypeScript: createUploadLink.mjs 'paths' is declared but its value is never read.ts(6133)

### ISSUE THREAD

**Issue by MortenEmde** [jaydenseric/apollo-upload-client#317]
When adding the `"maxNodeModuleJsDepth": 10` for TypeScript compatibility of v18.0.0, a TS error in `createUploadLink.mjs` is possibly triggered.

`'paths' is declared but its value is never read.ts(6133)`

This happens on line 166
```js
files.forEach((paths, file) => {
        customFormDataAppendFile(form, String(++i), file);
      });
```

suggested fix:
```js
files.forEach((_paths, file) => {
        customFormDataAppendFile(form, String(++i), file);
      });
```

**Comment by jaydenseric** [comment]
That's a good suggestion, but it's surprising that is causing errors for you. What is the particular setting that causes something like that to be a TS error? I thought my config here was quite strict; stricture than people's projects are likely to be:

https://github.com/jaydenseric/apollo-upload-client/blob/f44feafbca1301409d56d1d511ad26aae1f58471/jsconfig.json#L6

In the mean time, are you able to work around the issue by setting [`skipLibCheck`](https://www.typescriptlang.org/tsconfig#skipLibCheck) to `true`?

**Comment by jaydenseric** [comment]
It's:

```json
{
  "compilerOptions": {
    "noUnusedParameters": true
  }
}
```

**Comment by jaydenseric** [comment]
Thanks for raising this issue @MortenEmde; the fix has been published in [v18.0.1](https://github.com/jaydenseric/apollo-upload-client/releases/tag/v18.0.1) 🚀
